_posts/2024-08-28-Data-Prepper-2.9.0-is-ready-for-download.md

```diff
@@ -0,0 +1,120 @@
+
+Additionally, Data Prepper now has a new `startsWith` function to determine if a string value starts with another string.
+
+## Default route
+
+Data Prepper has had sink routing since the 2.0 release.
+With this capability, pipeline authors can use Data Prepper expressions to route events to different sinks to meet their requirements.
+One challenge that pipeline authors had was how to handle events that did not match any existing routes.
+A common solution that teams found to work-around this was to have a route that was the inverse of other routes.
+But, this required that they copy the other conditions and invert it.
+This was difficult to handle and even harder to maintain.
+
+Now, Data Prepper supports a special route named `_default`.
+By applying this route to a sink, pipeline authors can ensure that events that do not match any other routes will go a default sink of their choosing.
+
+For example, consider a simple situation where you want to route front-end and back-end events to different sinks.
+You can define two sinks for these and define your routes.
+But, what if you receive events that do not match?
+The sample pipeline below shows you how you could handle events that do not match either the front-end or back-end routes.
+
+```
+routes:
+  - frontend: '/service == "front-end"`
+  - backend: '/service == "back-end"`
+sink:
+ - opensearch:
+      routes:
+         - front-end
+ - opensearch:
+      routes:
+         - back-end
+  - opensearch:
+       routes:
+          - _default
+```
+
+## Performance
+
+The Data Prepper maintainers have been working toward improving the performance of Data Prepper.
+This release includes a number of internal improvements that speed up processing for many processors.
+You don't have to do anything other than update your version to get these improvements.
+
+Data Prepper 2.9 also offers some new features that you can use to help reduce out-of-memory errors or circuit breaker trips.
+Many pipelines involve taking some source data in the form of a string and then extracting that data into a structure.
+Some examples of these are `grok` and `parse_json`.
+When you use these processors, you more than double the size of each event you are processing.
+As the events flowing through the system are the largest part of memory usage, this will greatly increase your memory requirements.
+
+Many pipeline authors will use these processors and then in a second processor remove the source data.
+This is good, but it doesn't always make the available available for garbage collection when you need it.
+The reason is that Data Prepper pipelines operate on batches of data.
+As these batches of data move through the pipeline, the pipeline will expand the memory usage in one processor and then attempt to reduce it in the next.
+Because these happen in batches, Data Prepper may expand many thousands of events, before starting to remove the source data.
+
```

Comment:
Line above: "these happen" => Either "this happens" or name the noun represented by "these".

Reply:
"Because the memory expansion happens"